napari/_qt/qt_main_window.py

```diff
@@ -1557,16 +1597,62 @@ def _screenshot(
                     add_flash_animation(self._qt_viewer._welcome_widget)
             finally:
                 # make sure we always go back to the right canvas size
-                if size is not None or scale is not None:
+                if size is not None or scale is not None or fit_to_data_extent:
                     canvas.size = prev_size
+                if fit_to_data_extent:
+                    camera.center = old_center
+                    camera.zoom = old_zoom
         else:
             img = self._qt_window.grab().toImage()
             if flash:
                 add_flash_animation(self._qt_window)
         return img
 
+    def export_view(
+        self,
+        path=None,
+        scale=None,
+        flash=True,
+    ):
+        """Take currently displayed canvas, resets the view and create a screenshot without margins around the data.
+
+        Parameters
+        ----------
+        path : str
+            Filename for saving screenshot image.
+        scale : float
+            Scale factor used to increase resolution of canvas for the screenshot. By default, the currently displayed resolution.
+            Only used if `canvas_only` is True.
+        flash : bool
+            Flag to indicate whether flash animation should be shown after
+            the screenshot was captured.
+            By default, True.
+
+        Returns
+        -------
+        image : array
+            Numpy array of type ubyte and shape (h, w, 4). Index [0, 0] is the
+            upper-left corner of the rendered region.
+        """
+        img = QImg2array(
+            self._screenshot(
+                scale=scale,
+                flash=flash,
+                canvas_only=True,
+                fit_to_data_extent=True,
+            )
+        )
+        if path is not None:
+            imsave(path, img)
+        return img
+
     def screenshot(
-        self, path=None, size=None, scale=None, flash=True, canvas_only=False
+        self,
+        path=None,
+        size=None,
+        scale=None,
+        flash=True,
+        canvas_only=False,
```

Comment:
remove the final comma to undo these changes

Reply:
I think all these comments are now resolved:)